facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014-2019 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.facade.providers;
+
+import org.assertj.core.api.Condition;
+import org.jboss.pnc.dto.Environment;
+import org.jboss.pnc.dto.response.Page;
+import org.jboss.pnc.enums.SystemImageType;
+import org.jboss.pnc.facade.validation.DTOValidationException;
+import org.jboss.pnc.model.BuildEnvironment;
+import org.jboss.pnc.spi.datastore.repositories.BuildEnvironmentRepository;
+import org.jboss.pnc.spi.datastore.repositories.api.Repository;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.fail;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EnvironmentProviderTest  extends AbstractProviderTest<BuildEnvironment>{
+
+    @Mock
+    private BuildEnvironmentRepository repository;
+
+    @Spy
+    @InjectMocks
+    private EnvironmentProviderImpl provider;
+
+    @Override
+    protected AbstractProvider provider() {
+        return provider;
+    }
+
+    @Override
+    protected Repository<BuildEnvironment, Integer> repository() {
+        return repository;
+    }
+
+    BuildEnvironment env = prepareBuildEnvironment("Crash Test Dummy 1");
+
+    @Before
+    public void fill() {
+        List<BuildEnvironment> envs = new ArrayList<>();
+        envs.add(prepareBuildEnvironment("Hello"));
+        envs.add(prepareBuildEnvironment("My name is"));
+        envs.add(prepareBuildEnvironment("S-S-S-Slim Shady"));
+        envs.add(env);
+        fillRepository(envs);
+    }
+
+    @Test
+    public void testGetAll() {
+        Page<Environment> all = provider.getAll(0, 10, null, null);
+
+        assertThat(all.getContent())
+                .hasSize(4)
+                .haveExactly(1, new Condition<>(e -> e.getName().equals(env.getName()), "Environment present"));
+    }
+
+    @Test
+    public void testStore() {
```

Comment:
I think we don't allow storing of environments via REST.

Reply:
No, but other providers may use it I guess?

Reply:
I'll update the provider to UnsupportedOpException